string req
float64 goal
--- 
bool res
